apps/emqx/src/emqx_schema.erl

```diff
@@ -1093,13 +1095,42 @@ to_bar_separated_list(Str) ->
 to_ip_port(Str) ->
     case string:tokens(Str, ":") of
         [Ip, Port] ->
-            case inet:parse_address(Ip) of
+            case get_addr(Ip) of
                 {ok, R} -> {ok, {R, list_to_integer(Port)}};
                 _ -> {error, Str}
             end;
         _ -> {error, Str}
     end.
 
+get_addr({_1, _2, _3, _4} = IPV4) ->
+    {ok, IPV4};
+
+get_addr({_1, _2, _3, _4, _5, _6, _7, _8} = IPV6) ->
+    {ok, IPV6};
+
+get_addr(Domain) when is_list(Domain) ->
+    Trys = case inet_db:res_option(inet6) of
+               false ->
+                   [inet, inet6];
+               _ ->
+                   [inet6, inet]
+           end,
+    try_get_addrs(Trys, Domain);
+
+get_addr(Any) ->
+    {error, {wrong_address, Any}}.
+
+try_get_addrs([Family | T], Domain) ->
```

Comment:
I think we should not convert the domain name to an IP here. This will force us to bind to this IP and cause some external resource load balancing strategies to fail.